### PR TITLE
Prevent ZeroDivisionError when devicePixelRatio() returns 0

### PR DESCRIPTION
**Pull request by AlexHarn**
This bug only occurs under very special circumstances. For some reason when using Screen Scaling in KDE while also downscaling the terminal emulator Konsole, in which matplotlib is run, devicePixelRatio() returns 0 and causes a ZeroDevisionError. I do not know other circumstances under which this bug occurs. It is also probably not really a bug in matplotlib, but this fixes the issue for me.

You may ask why this setup is necessary to begin with, the answer can be found [here](https://www.reddit.com/r/kde/comments/5u44r3/konsole_screen_tearing_workaround/). Essentially it is a workaround necessary to prevent screen tearing in Konsole for some very few people including myself.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jkseppan** [comment]
I'm not sure I understand the underlying issue, but it does seem that a ratio of 0 makes no sense here, so returning 1 is better.

File: lib/matplotlib/backends/backend_qt5.py [hunk 1]
```diff
@@ -269,7 +269,11 @@ def _update_figure_dpi(self):
     def _dpi_ratio(self):
         # Not available on Qt4 or some older Qt5.
         try:
-            return self.devicePixelRatio()
+            ratio = self.devicePixelRatio()
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/backends/backend_qt5.py]: a comment would be nice, indeed.

File: lib/matplotlib/backends/backend_qt5.py [hunk 2]
```diff
@@ -268,12 +268,9 @@ def _update_figure_dpi(self):
     @property
     def _dpi_ratio(self):
         # Not available on Qt4 or some older Qt5.
-        try:
-            ratio = self.devicePixelRatio()
-            if ratio == 0:
-                return 1
-            else:
-                return ratio
+            try:
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_qt5.py]: Indentation error

Comment (AlexHarn) [reply lib/matplotlib/backends/backend_qt5.py]: I'm sorry, I should have double checked that...

**Comment by tacaswell** [comment]
@meeseeksdev backport to v2.2.x

**Pull request merged**